caikit/runtime/utils/servicer_util.py

```diff
@@ -211,209 +212,70 @@ def validate_data_model(
-                            field_name,
-                            field_value,
-                            type(field_value),
-                        )
-                        # Start by getting the class name for this particular
-                        # field (e.g., RawDocument)
-                        class_name = type(field_value).DESCRIPTOR.name
-
-                        # special case for model pointer
-                        if class_name == "ModelPointer":
-                            log.debug2("field_value is a ModelPointer obj")
-                            if field_value.model_id:
-                                model_manager = ModelManager.get_instance()
-                                model_retrieved = model_manager.retrieve_model(
-                                    field_value.model_id
-                                )
-                                caikit_library_request_dict[
-                                    field_name
-                                ] = model_retrieved
-                        else:
-                            # Now get the Caikit Library CDM class of the same
-                            # name
-                            caikit_library_class = getattr(cdm, class_name)
-                            # Use the Caikit Library CDM class's from_proto
-                            # method to turn our protobufs field message into an
-                            # instance of the Caikit Library CDM class
-                            instance = caikit_library_class.from_proto(field_value)
-                            # Add to the request dictionary, using the message
-                            # field's name as the key (since, by convention, the
-                            # argument name to the module run function will be
-                            # the same as the field name)
-                            caikit_library_request_dict[field_name] = instance
-
-        log.debug2(
-            "caikit_library_request_dict returned is: %s", caikit_library_request_dict
-        )
-
-        return caikit_library_request_dict
+                # Remove empty iterables since we cannot distinguish between
+                # unset and empty repeated fields
+                field_value = getattr(request, field.name)
+                if isinstance(field_value, Iterable) and len(field_value) == 0:
+                    unset_field_names.append(field.name)
+        for unset_field_name in unset_field_names:
+            if unset_field_name in kwargs_dict:
+                kwargs_dict.pop(unset_field_name)
+
+        # 2. Remove any fields not in the module signature
+        absent_field_names = [
+            field
+            for field in kwargs_dict.keys()
+            if field not in module_signature.parameters.keys()
+        ]
+        for absent_field_name in absent_field_names:
+            if absent_field_name in kwargs_dict:
```

Comment:
is this a redundant check? The `absent_field_names` list is populated by iterating through all `fields` in `kwargs_dict.keys()` right?